### DISCUSSION TITLE
Github Action `on: create` not trigerring

### DISCUSSION THREAD

**Discussion by garronej** [community/community#24955] [Actions]
<p>Hello there,</p>
<p>I have a project under the form of a template that is supposed to trigger a GitHub Actions workflow when the user click on “use this template”</p>
<p><a href="https://github.com/garronej/ts_ci/blob/ccbbcf384d3bc5a4867fdca50fb91762c9bd103d/.github/workflows/template_initialization.yaml" rel="noopener nofollow ugc">This is the Workflow in question.</a></p>
<p>It use to work great but now, more often than not when a user click on “use this template” the workflow <code>template_initialization.yaml</code> doesn’t trigger.</p>
<p>Thank you for your help and for the otherwise amassing services.</p>
<p>Best regards</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/garronej">@garronej</a>,</p>
<p>I can reproduce the same problem.</p>
<p>According to my tests, if the workflow runs on the <strong>create</strong> event is never triggered on the template repository, when creating a new repository with the template repository, the workflow will not be triggered on the new repository.</p>
<p>However, after I triggered the workflow at least once on the template repository, the problem no longer occurs. When I tried again to create a new repository with the template repository, the workflow can be triggered as expected on the new template repository.</p>

**Comment by Simran-B** [comment]
<p>Could this be somehow related to the changed default branch name (master &gt; main)?</p>

**Comment by garronej** [comment]
<p><a class="mention" href="/brightran">@brightran</a> THANK YOU, THANK YOU, THANK YOU.<br>
For taking the time to investigate. It works now.</p>
<p><a class="mention" href="/simran-b">@Simran-B</a> Thank you for your answer as well. I had migrated to using <code>develop</code> instead of <code>master</code> a while ago in anticipation.</p>

**Comment by garronej** [comment, marked as answer]
<p>Well after many tries and many errors I finally gave up on using on create but I found a workaround.</p>
<p>More details:</p><aside class="quote" data-post="4" data-topic="160641">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/github-action-on-create-not-trigerring-again/160641/4">Github Action `on: create` not trigerring (again)</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Well in the end, after hours of tries and error I got to the conclusion that for a workflow to run after a template creation it must have ran successfully on the last commit made on the source repo. 
So I can’t use  on create but I have fond a workaround.
  </blockquote>
</aside>
